sample-apps/go-sample-app/collection/http_traces.go

```diff
@@ -0,0 +1,104 @@
+package collection
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/bitly/go-simplejson"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Contains all of the endpoint logic.
+var tracer = otel.Tracer("sample-app")
+
+// AwsSdkCall mocks a request to s3. ListBuckets are nil so no credentials are needed.
+// Generates an Xray Trace ID.
+func (rqmc *requestBasedMetricCollector) AwsSdkCall(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	s, err := session.NewSession(&aws.Config{
+		Region: aws.String("us-west-2")},
+	)
+
+	svc := s3.New(s)
+	svc.ListBuckets(nil) // nil or else would need real aws credentials
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	_, span := tracer.Start(
```

Comment:
```suggestion
	ctx, span := tracer.Start(
```

Reply:
This change overwrites the context thats passed from the request

Reply:
nvm, i fixed this. the suggestion was right